SoC/gd32vf103/Common/Source/system_gd32vf103.c

```diff
@@ -185,7 +185,7 @@ static void system_clock_108m_hxtal(void)
     \param[out] none
     \retval     none
 */
-static void system_clock_config(void)
+void system_clock_config(void)
 {
```

Comment:
Currently make it external function may not solve the situation that you want to make it customizable, maybe weak implementation is a better solution, and you can also reinitialize clock it in application code.

Reply:
I didn't make it extern for customization, but for having a simple way to re-initialize the clock after waking up from deep sleep.

I mean, I want to use it like this in application code:

```
pmu_to_deepsleepmode(PMU_LDO_NORMAL, WFI_CMD);
system_clock_config();
SystemCoreClock = get_cpu_freq();
```

I think this is a common use case, because the deep sleep mode sets the system clock to IRC8M - and most users will run the device with the default 108 MHz PLL system clock which, usually, they want to continue to use after returning from deep sleep.

---

Perhaps it even makes more sense to simply add

```
system_clock_config();
SystemCoreClock = get_cpu_freq();
```

at the end of `pmu_to_deepsleepmode()` such that a user really only has to call

```
pmu_to_deepsleepmode(PMU_LDO_NORMAL, WFI_CMD);
```

and be done with it. But for that to work `system_clock_config()` would also need to be extern.

What do you think?